src/main/java/com/annapolisworks/ExpressionParser.java

```diff
@@ -0,0 +1,65 @@
+package com.annapolisworks;
+
+public class ExpressionParser {
+
+    private int mIndex;
+    char c;
+    double firstNumber;
+    double secondNumber;
+    Operator thisOperator;
+    OpNode firstNode;
+    OpNode thisNode;
+    private String expr;
+
+    public OpNode parse(String expressionToParse) {
+        //need to add provision for 2.3.4 + 1 <- 2 decimals
+        //need to add provision for 2+-5 <- would simply read over first operator
+
+        expr = expressionToParse;
+        //create a 0+? first node to be overwritten Or in case user enters a single number
+        thisOperator = Operator.makeOperator('+');
+        firstNumber = 0d;
+
+        mIndex = 0;
+        while(mIndex < expr.length()){
```

Comment:
space after while. Should be `while (`. You also forgot your space after the expression. Space between `){`. Should be:

`while (mIndex < expr.length()) {`